### PR TITLE
[tempo-distributed] Set registry for Memcached, Memcached Exporter and Gateway services

### PR DESCRIPTION
**Pull request by aravandi**
Signed-off-by: Amir <amirghotbi@gmail.com>

Hi,
This PR helps to be able to set a registry for Memcached, Memcached-Exporter, and Gateway services.

### CODE REVIEW

File: charts/tempo-distributed/templates/memcached/_helpers-memcached.tpl [hunk 1]
```diff
@@ -20,3 +20,19 @@ memcached selector labels
 {{ include "tempo.selectorLabels" . }}
 app.kubernetes.io/component: memcached
 {{- end }}
+
+{{/*
+memcached image
+*/}}
+{{- define "tempo.memcachedImage" -}}
+{{- $dict := dict "service" .Values.memcached.image "global" .Values.global.image "tempo" (dict) -}}
```
Comment (aravandi) [inline charts/tempo-distributed/templates/memcached/_helpers-memcached.tpl:28]: The `tempoImage` template is looking for a tempo key in the provided dict, So, we could provide it with an empty dict.

**Comment by joe-elliott** [comment]
We have been having issues with our CLA integration lately, but it seems like it is working now? You will need to sign the CLA before we can merge.

File: charts/tempo-distributed/templates/memcached/_helpers-memcached.tpl (hunk 1 above)
Comment (aravandi) [reply charts/tempo-distributed/templates/memcached/_helpers-memcached.tpl:28]: Let me describe it a little bit more here. from our last conversation, we decided to remove the `tempo` key from the `dict` object. 
I did that change then I got an error:
```text
UPGRADE FAILED: 
  template: tempo-distributed/templates/memcached/statefulset-memcached.yaml:41:20: 
  executing "tempo-distributed/templates/memcached/statefulset-memcached.yaml" at 
    <include "tempo.memcachedImage" .>: 
       error calling include: 
         template: tempo-distributed/templates/memcached/_helpers-memcached.tpl:29:4:
         executing "tempo.memcachedImage" at 
         <include "tempo.tempoImage" $dict>: error calling include: 
           template: tempo-distributed/templates/_helpers.tpl:31:67: 
           executing "tempo.tempoImage" at <.tempo.registry>: nil pointer evaluating interface {}.registry
```
The error says the `tempo.tempoImage` template in the `tempo-distributed/templates/_helpers.tpl` file needs a `tempo` key to find the `tempo.registry` 
in the given object (dict) but the `tempo` key has been removed.
As a resolution, we have to provide the `tempo` key in the `dict` object, and it has to be an empty object; hence I added the `tempo (dict)` to the `dict` object.

I hope that helped you to get to the point why the `tempo` key has to be set there.

Comment (joe-elliott) [reply charts/tempo-distributed/templates/memcached/_helpers-memcached.tpl:28]: awesome, thank you!

**Review by joe-elliott — APPROVED** [review]
Nice addition. Thanks!

**Pull request merged**